### ISSUE TITLE
Include "Block HTML canvas and WebGL fingerprinting" into Bravery Panel

### ISSUE THREAD

**Issue by luixxiul** [brave/browser-laptop#1876]
**Describe the issue you encountered:** Sometimes it is hard to detect the issue, when the site does not work as it should, because of blocking the HTML canvas and WebGL fingerprinting. In my case, adding a macro at Zendesk.

**Expected behavior:** Display a toggle to enable/disable them, next to the "Block phising / malware"

- Platform (Win7, 8, 10? OS X? Linux distro?): Windows 10

- Brave Version: 0.10.0

- Any related issues: #1621

If it is hard to difficult to implement it like this (blocking them needs to be a global config, for example) , something which indicates that they were blocked, like the NoScript icon, would help users to understand what is going on. Currently they have to open devtool and check out the output on the console.

**Comment by bbondy** [comment]
hah, I was just asking about this same thing!  I think ti's a great idea too.

**Comment by diracdeltas** [comment]
Might be good to reword this as `Block advanced fingerprinting methods` so it can encompass things like battery status API and audio fingerprinting in the future

**Comment by diracdeltas** [comment]
CC @bradleyrichter for design: Should there be a number in the bravery panel for # of fingerprinting attempts that were blocked on the page? Similar to #1833

**Comment by diracdeltas** [comment]
copying my comments from slack re: design issues related to this:

1. should there be separate switches for canvas/webgl/audio/webrtc/battery/etc fingerprinting? or just one switch for all detectable browser fingerprinting methods? if one switch, there should be a tooltip or something for advanced users to see what methods are being blocked.
2. should there be a number that shows # of fingerprinting methods blocked? it makes sense but maybe the panel would get too wide.
3. originally i was going to show a popup notification message for when Site X is potentially trying to fingerprint you via one of these methods, so the user can choose Allow/Deny. this might be redundant now that there is a way to easily see the list of fp methods and easily allow/deny in the Bravery panel. i am leaning against showing these notifications because i think they are somewhat annoying, although Tor Browser does.

IMO the fingerprinting switch should be off-by-default in regular browsing mode and on-by-default for private tabs.

**Comment by diracdeltas** [comment]
merging into https://github.com/brave/browser-laptop/issues/1951
